utils/mensaUtils.py

```diff
@@ -22,20 +34,77 @@ def get_mensa_plan(date: datetime) -> list[Meal]:
     Returns:
         list[Meal]: A list of Meal objects representing the meals available on the given date.
     """
-    page = requests.get(f"{Constants.URLS.MENSAPLAN}{date.strftime('%Y-%m-%d')}")
-    soup = BeautifulSoup(page.content, 'html.parser')
+    page = requests.get(
+        f"{Constants.URLS.MENSAPLAN}{date.strftime('%Y-%m-%d')}"
+    )
+    soup = BeautifulSoup(page.content, "html.parser")
+
+    meals: list[Meal] = []
+
+    for meal_element in soup.select(".type--meal"):
+        meal_type_element = meal_element.select_one(".meal-tags span")
+        meal_price_element = meal_element.select_one(".meal-prices span")
+
+        if not meal_type_element or not meal_price_element:
+            continue
+
+        meal_type = MealType(meal_type_element.text)
+        meal_price = Price.get_from_string(meal_price_element.text.strip())
+
+        if meal_type is not MealType.PASTA:
+            meal = retrieve_standard_meal_data(
+                meal_type,
+                meal_element,
+                meal_price
+            )
+            if meal:
+                meals.append(meal)
+            continue
+
+        for pasta_element in meal_element.select(".meal-subitem"):
+            meal = extract_pasta_meal_data(meal_type, pasta_element, meal_price)
+            if meal:
+                meals.append(meal)
 
-    meals = []
+    return meals
```

Comment:
Just a thought, would it come in handy to use the `yield` keyword here for returning the list of meals?

Reply:
Ist gemacht!